index.js

```diff
@@ -178,18 +202,52 @@ exports.generate = async (inputFile, outputDirectory, isApiMonolith, userProvide
 				if (schema.allOf) {
 					schema = unifyModel(allSchemas, schema.allOf);
 				}
-				return {
-					MODEL_NAME: pascalCase(schemaName),
-					MODEL_DESCRIPTION: schema.description,
-					MODEL_PROPERTIES: _.map(schema.properties, (property, propertyName) => {
+
+				let properties;
+
+			
+				if (schema.type === "array" && schema.items) {
```

Comment:
I think overall the logic is sound, but i see a bit too much code duplication. Extracting out the property "singleton" would be the first to simplify

Reply:
DRY'ed up the object property creation to a helper function